muikku-school-data-pyramus/src/main/java/fi/otavanopisto/muikku/plugins/schooldatapyramus/PyramusUserSchoolDataBridge.java

```diff
@@ -1200,7 +1202,94 @@ public BridgeResponse<List<OrganizationContactPerson>> listOrganizationContactPe
   private StudentStudyPeriod[] listStudentStudyPeriods(Long pyramusStudentId) {
     return pyramusClient.get(String.format("/students/students/%d/studyPeriods", pyramusStudentId), StudentStudyPeriod[].class);
   }
+  
+  @Override
+  public BridgeResponse<List<StudentContactLogEntryRestModel>> listStudentContactLogEntriesByStudent(SchoolDataIdentifier userIdentifier){
+    Long pyramusStudentId = identifierMapper.getPyramusStudentId(userIdentifier.getIdentifier());
 
+    if (pyramusStudentId != null) {
+      BridgeResponse<StudentContactLogEntryRestModel[]> response = pyramusClient.responseGet(String.format("/students/students/%d/contactLogEntries", pyramusStudentId), StudentContactLogEntryRestModel[].class);
+      List<StudentContactLogEntryRestModel> contactLogEntries = null;
+      if(response.getEntity() != null) {
+        contactLogEntries = new ArrayList<>();
+        for (StudentContactLogEntryRestModel contactLogEntry : response.getEntity()) {
+          contactLogEntries.add(contactLogEntry);
+        }
+      }
+      return new BridgeResponse<List<StudentContactLogEntryRestModel>>(response.getStatusCode(), contactLogEntries);
+    }
+    logger.warning(String.format("PyramusUserSchoolDataBridge.listStudentContactLogEntriesByStudent malformed user identifier %s\n%s",
+      userIdentifier,
+      ExceptionUtils.getStackTrace(new Throwable())));
+    throw new SchoolDataBridgeInternalException(String.format("Malformed user identifier %s", userIdentifier));
+  }
+  
+  @Override 
+  public BridgeResponse<StudentContactLogEntryRestModel> createStudentContactLogEntry(SchoolDataIdentifier studentIdentifier, StudentContactLogEntryRestModel payload){
+    Long studentId = identifierMapper.getPyramusStudentId(studentIdentifier.getIdentifier());
+    
+    if (studentId == null) {
+      logger.severe(String.format("Student for identifier %s not found", studentIdentifier));
+      return null;
+    }
+    return pyramusClient.responsePost(String.format("/students/students/%d/contactLogEntries", studentId), Entity.entity(payload, MediaType.APPLICATION_JSON), StudentContactLogEntryRestModel.class);
```

Comment:
The creatorId in all response objects (contact log entries and their comments) are ids of the users in Pyramus. Before handing the response over to Muikku front-end, they should be converted to ids of the users in Muikku (i.e. userEntityId). Something like...

```
if (response.getEntity() != null && response.getEntity().getCreatorId() != null) {
  SchoolDataIdentifier identifier = identifierMapper.getStaffIdentifier(response.getEntity().getCreatorId());
  if (identifier != null) {
    UserSchoolDataIdentifier usdi = userSchoolDataIdentifierController.findUserSchoolDataIdentifierBySchoolDataIdentifier(identifier);
    if (usdi != null) {
      response.getEntity.setCreatorId(usdi.getUserEntity.getId());
    }
    else {
      // TODO log warning
      response.getEntity.setCreatorId(null);
    }
  }
  else {
    // TODO log warning
    response.getEntity.setCreatorId(null);
  }
}
// TODO Do the same with comments
```

Naturally this applies to all methods returning contact log entries and/or their comments.